src/server/zset_family.cc

```diff
@@ -2060,6 +2076,75 @@ void ZRemRangeGeneric(string_view key, const ZSetFamily::ZRangeSpec& range_spec,
   }
 }
 
+// Returns the key of the first non empty set found in the list of shard arguments.
+// Returns nullopt if none.
+std::optional<std::string> GetFirstNonEmptyKeyFound(EngineShard* shard, Transaction* t) {
```

Comment:
I believe you can return string_view from GetFirstNonEmptyKeyFound
The ShardArgs are pointing to the command args that are allocated when parsing the command

Reply:
Done